golang/pkg/ray/task_executor.go

```diff
@@ -0,0 +1,83 @@
+package ray
+
+/*
+   #cgo CFLAGS: -I../../../ray/src/ray/core_worker/lib/golang
+   #cgo LDFLAGS: -shared  -L/root/ray/bazel-bin/ -lcore_worker_library_go -lstdc++
+   #include <string.h>
+   #include "go_worker.h"
+*/
+import "C"
+import (
+    "fmt"
+    "reflect"
+    "unsafe"
+
+    "github.com/ray-project/ray-go-worker/pkg/ray/generated"
+    "github.com/ray-project/ray-go-worker/pkg/util"
+    "github.com/vmihailenco/msgpack/v5"
+)
+
+var actor interface{}
+var actorReflectValue reflect.Value
+
+//export go_worker_execute
+func go_worker_execute(taskType int, rayFunctionInfo []*C.char, args []C.struct_DataBuffer, returnValue []*C.struct_ReturnValue) {
+    if taskType == int(generated.TaskType_ACTOR_CREATION_TASK) {
+        d := (*reflect.SliceHeader)(unsafe.Pointer(&rayFunctionInfo))
+        util.Logger.Debugf("slice info:%v", *d)
+        goTypeName := C.GoString(rayFunctionInfo[0])
+        goType, ok := typesMap[goTypeName]
+        if !ok {
+            panic(fmt.Errorf("type not found:%s", goTypeName))
+        }
+        actor = reflect.New(goType).Interface()
+        actorReflectValue = reflect.ValueOf(actor)
+        util.Logger.Debugf("created actor for:%s", goTypeName)
+    } else if taskType == int(generated.TaskType_ACTOR_TASK) {
+        methodName := C.GoString(rayFunctionInfo[0])
+        util.Logger.Debugf("invoke %s", methodName)
+        methodValue := actorReflectValue.MethodByName(methodName)
```

Comment:
Is it possible to avoid runtime reflection, by e.g., building up a method pointer table on actor creation? Not sure if the overhead matters here though.

Reply:
I have added `actorMethodMap` to save method pointer. It can avoid to  iterate the method list.